### PR TITLE
Allow responses only in DM with bot

### PR DESCRIPTION
**Pull request by ryanpapetti**
I, unsurprisingly, introduced quite a few bugs when making the bot respond in an IM (DM with bot). This PR fixes them by doing the following:

- Adds a feature flag (`FEATURE_RESPOND_VIA_MENTION`) that turns off the bot from responding via a mention in a normal channel
  - if the bot is tagged in a channel for some reason, then it'll respond with a message saying it can't help
  - if it is tagged in an IM, it still responds as normal

- Removes all other shortcut / template features to reduce complexity
- Forces the IM to properly recognize logic, where it hadn't before

ALSO, it turns out the Slack related tokens are different than what was in the container. We need to change these ( I have the new ones )

### CODE REVIEW

File: app/bolt_listeners.py [hunk 1]
```diff
@@ -66,121 +62,156 @@ def respond_to_app_mention(
...
+                    openai_api_key=context.get("OPENAI_API_KEY"),
+                    context=context,
+                    text=NO_APP_MENTION_MESSAGE,
+                ),
             )
-            return
+    
+    elif payload.get('channel_type') == "im":
+        return respond_in_dm(context=context, payload=payload, client=client, logger=logger)
+
```
Comment (ryanpapetti) [inline app/bolt_listeners.py:79]: This ensures that the bot tells the user it's unable to respond via an app mention.

Comment (ryanpapetti) [inline app/bolt_listeners.py:79]: This ensures that the bot tells the user it's unable to respond via an app mention.

File: app/bolt_listeners.py [hunk 2]
```diff
@@ -228,12 +240,16 @@ def respond_in_dm(
...
+    wip_reply = None
     try:
         openai_api_key = context.get("OPENAI_API_KEY")
 
+        correct_ts = 'thread_ts' if 'thread_ts' in payload else 'ts'
+
         messages_in_context = client.conversations_replies(
                 channel=context.channel_id,
-                ts=payload.get('thread_ts'),
+                ts=payload[correct_ts],
```
Comment (ryanpapetti) [inline app/bolt_listeners.py:252]: Apparently, trying to "hack" the `thread_ts` didn't work and I needed to use the correct timestamp

File: app/bolt_listeners.py [hunk 3]
```diff
@@ -411,826 +424,21 @@ def respond_to_new_message(
...
-    }
-    if is_error is True:
-        del view["submit"]
-
-    client.views_open(
-        trigger_id=body.get("trigger_id"),
-        view=view,
-    )
-    ack()
-
```
Comment (ryanpapetti) [inline app/bolt_listeners.py:756]: This is one example of a function (a shortcut) that we won't use, so it makes no sense to keep it in the listeners for now

Comment (ryanpapetti) [inline app/bolt_listeners.py:756]: This is one example of a function (a shortcut) that we won't use, so it makes no sense to keep it in the listeners for now

File: app/bolt_listeners.py [hunk 4]
```diff
@@ -411,826 +424,21 @@ def respond_to_new_message(
...
-                        "text": {
-                            "type": "mrkdwn",
-                            "text": f"{text}\n\n:warning: My apologies!"
-                            f"An error occurred while generating the summary of this thread: {e}",
-                        },
-                    },
-                ],
-            },
-        )
-
```
Comment (ryanpapetti) [inline app/bolt_listeners.py:1197]: Here is all of the code that needs to be removed (over 500 lines!!!!!)

**Review by fdebijl — APPROVED** [review]
Confuscius says: a junior developer adds code, a medior developer refactors code and a senior developer removes code

**Pull request merged**